docs/markdown/Contributions/development/contributor-setup.md

```diff
@@ -28,6 +28,11 @@ Step 2: Bootstrap the Rust engine
 
 Pants uses Rustup to install Rust. Run the command from <https://rustup.rs> to install Rustup; ensure that `rustup` is on your `$PATH`.
 
+You also need a suitable Python 3.9 interpreter. The [Python Standalone Builds](https://gregoryszorc.com/docs/python-build-standalone/main/) by Gregory Szorc should work.
+Avoid the Python interpreters from Ubuntu's deadsnakes PPA, they don't work with Pants. The python39 package from the Nix package manager has also caused problems for users in the past.
```

Comment:
Maybe we could generalize this: 
```suggestion
If your system Python is not the version Pants expects (currently Python 3.9), you'll need to provide one.  Python interpreters from Linux or Mac distributions sometimes have quirks that can cause headaches with bootstrapping the dev venv.  Some examples of Pythons that work well with Pants are those provided by:

- [Fedora](https://packages.fedoraproject.org/pkgs/python3.9/python3.9/)
- [ASDF](https://github.com/asdf-community/asdf-python)
- [PyEnv](https://github.com/pyenv/pyenv)

Providers that sometimes cause issues include:

- Ubuntu Deadsnakes
- System MacOS Python
```

Reply:
@jriddy 
Regarding Nix, I think I must have been holding it wrong. I've managed to get it to work just fine with the Nix Python interpreter, see github.com/pantsbuild/pants/pull/19586. So I suggest we drop the line about Nix.

Reply:
That said, I think you have worded it better than I have, so I've added your suggestion in a new commit. Thanks 😃